command/genesis/polybft_params.go

```diff
@@ -235,20 +271,27 @@ func (p *genesisParams) deployContracts() (map[types.Address]*chain.GenesisAccou
 	return allocations, nil
 }
 
-func generateExtraDataPolyBft(validators []GenesisTarget) []byte {
+// generateExtraDataPolyBft populates Extra with specific fields required for polybft consensus protocol
+func generateExtraDataPolyBft(validators []*polybft.Validator, publicKeys []*bls.PublicKey) ([]byte, error) {
+	if len(validators) != len(publicKeys) {
+		return nil, fmt.Errorf("expected same length for genesis validators and BLS public keys")
+	}
+
 	delta := &polybft.ValidatorSetDelta{
 		Added:   make(polybft.AccountSet, len(validators)),
 		Removed: bitmap.Bitmap{},
 	}
 
 	for i, validator := range validators {
 		delta.Added[i] = &polybft.ValidatorMetadata{
-			Address: types.Address(validator.Account.Ecdsa.Address()),
-			BlsKey:  validator.Account.Bls.PublicKey(),
+			Address: validator.Address,
+			BlsKey:  publicKeys[i],
+			// convert from wei to token amount (voting power == tokens amount)
+			VotingPower: validator.Balance.Div(validator.Balance, big.NewInt(polybft.WeiScalingFactor)).Uint64(),
```

Comment:
Could you extract voting power scaling to a separate function?

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/904/commits/8c7c342c70b9c63d3e47b79173689704f4361b7f